sdk/core/core/test/cmocka/CMakeLists.txt

```diff
@@ -20,23 +20,14 @@ endif()
 add_cmocka_test(${TARGET_NAME} SOURCES
```

Comment:
I am not a fan of having all tests in a /cmocka/ directory (seems unnecessary). Maybe we should remove those (**as a separate PR**). What was the reason to structure the tests this way in another nested subdirectory?

Reply:
reason was Rick asking cmocka test to be in one folder and any other (we used to have the raw in-house test) test out of it. Think this is a way to say that any test in there should be cmocka only.
I never really right back to Rick about it.  usually folder structure is not as relevant, so I just did as he asked XD

Reply:
@gilbertw - can we get rid of the intermediate `cmocka` directories in all our tests. Have another directory level/indirection when navigating is cumbersome and I am not sure what it buys us. Is there a benefit I am not seeing?